### DISCUSSION TITLE
Theme Generator like Shadcn

### DISCUSSION THREAD

**Discussion by austinm911** [jpudysz/react-native-unistyles#204] [Feature Requests]
Hey! New to the RN ecosystem playing with a project and using Unistyles for the first time. 

I found this [theme file](https://gist.github.com/TheUltDev/47c68faee67b997425aa5b76911f4361) (from [this tweet](https://x.com/theultdev/status/1785470538342948865)) really helpful as a starting point. 

It would probably help greatly if there was a theme selector built for users' to customize themes and generate a theme.ts like shadcn-theme sites allow.

I'm used to the tailwind style of things and maybe it would be helpful to provide default colors like tailwind does?

Anyways, thanks for the package

See:
[shadcn Themes](https://ui.shadcn.com/themes)
[1000+ shadcn themes](https://ui.jln.dev/)

**Comment by jpudysz** [comment]
Hey! 👋 

Yeah TheUltDev is working on the components library for the Unistyles. He is posting his progress on the Unistyles Discord.

Anyway, moving back to your suggestion. I'm intentionally didn't include any colors, breakpoints or components. The idea of Unistyles is to be extensible and to build own design system, rather than using one implemented by me. 

Building such design system is super easy. You just need to specify breakpoints and theme colors / values. Later you can work on your components. 

I understand the value of shadcn type of library (I'm even using shadcn for my personal project), but I'm not willing to build one for Unistyles. 

The main factor is time and lack of skills regarding designing UI. I want to focus on core of Unistyles and extend it towards better DX.

I hope that answers your question. 

By the way! Marcin created layout system for the Unistyles. It can be also helpful to get started with some small addons: https://stacks.grapp.dev
